### PR TITLE
refactor: update `blas/ext/base/dsumkbn` to follow current project conventions

### LINKED ISSUE #1505: [RFC]: refactor `blas/ext/base/dsumkbn` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dsumkbn`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dsumkbn) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by headlessNode**
Resolves #1505 

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors `blas/ext/base/dsumkbn` to follow current project conventions.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1505 
-   progresses #1152 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by headlessNode** [comment]
@kgryte @Planeshifter Please review.

File: lib/node_modules/@stdlib/blas/ext/base/dsumkbn/benchmark/benchmark.js [hunk 1]
```diff
@@ -21,14 +21,19 @@
 // MODULES //
 
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require('@stdlib/random/base/uniform');
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsumkbn/benchmark/benchmark.js]: Please conform to project spacing conventions. You should be consistent with what you see.

Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsumkbn/benchmark/benchmark.js]: You are missing `.factory` from your import.

**Comment by headlessNode** [comment]
@kgryte I apologize for the oversight. I will make sure such oversight is avoided in the future. Also, I've added the suggested changes, please check.

**Review by Planeshifter — APPROVED** [review]
Thanks for your contribution, @headlessNode! Will merge shortly.

**Pull request merged**